tests/HUBViewModelDiffTests.m

```diff
@@ -184,10 +184,68 @@ - (void)testComplexChangeSet
     XCTAssert([diff.reloadedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:1 inSection:0]]);
-    XCTAssert([diff.reloadedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:7 inSection:0]]);
+    XCTAssert([diff.reloadedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:8 inSection:0]]);
     XCTAssert(diff.reloadedBodyComponentIndexPaths.count == 2);
     XCTAssert(diff.insertedBodyComponentIndexPaths.count == 2);
     XCTAssert(diff.deletedBodyComponentIndexPaths.count == 2);
 }
 
+- (void)testInsertionOfComponentModelAtStartWithDataChanges
+{
+    NSArray<id<HUBComponentModel>> *firstComponents = @[
+                                                        [self createComponentModelWithIdentifier:@"component-1" customData:nil],
+                                                        [self createComponentModelWithIdentifier:@"component-2" customData:nil],
+                                                        [self createComponentModelWithIdentifier:@"component-3" customData:nil]
+                                                        ];
+    id<HUBViewModel> firstViewModel = [self createViewModelWithIdentifier:@"Test"
+                                                               components:firstComponents];
+    NSArray<id<HUBComponentModel>> *secondComponents = @[
+                                                         [self createComponentModelWithIdentifier:@"component-0" customData:nil],
+                                                         [self createComponentModelWithIdentifier:@"component-1" customData:@{@"test": @1}],
+                                                         [self createComponentModelWithIdentifier:@"component-2" customData:@{@"test": @1}],
+                                                         [self createComponentModelWithIdentifier:@"component-3" customData:@{@"test": @1}]
+                                                         ];
+    id<HUBViewModel> secondViewModel = [self createViewModelWithIdentifier:@"Test"
+                                                                components:secondComponents];
+
+    HUBViewModelDiff *diff = [HUBViewModelDiff diffFromViewModel:firstViewModel toViewModel:secondViewModel];
+    XCTAssert(diff.reloadedBodyComponentIndexPaths.count == 3);
+    XCTAssert(diff.insertedBodyComponentIndexPaths.count == 1);
+    XCTAssert(diff.deletedBodyComponentIndexPaths.count == 0);
+    XCTAssert([diff.insertedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:0 inSection:0]]);
+    XCTAssert([diff.reloadedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:0 inSection:0]]);
+    XCTAssert([diff.reloadedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:1 inSection:0]]);
+    XCTAssert([diff.reloadedBodyComponentIndexPaths containsObject:[NSIndexPath indexPathForItem:2 inSection:0]]);
+}
+
+- (void)testInsertionOfComponentModelsAtStartWithDataChanges
```

Comment:
Suggestion: I had to spend quite a few minutes to figure out what the difference was between this test and the one above. Renaming it to `testInsertionOfMultipleComponentModelsAtStartWithDataChanges` would make it a bit more explicit (IMO).